README.md

```diff
@@ -118,3 +101,171 @@ Remote debugging using :2000
+
+The first configuration commands are ran from the root of your clone of the blackmagic repository
+``` sh
+cd /path/to/repository/blackmagic
+```
+
+All following commands are ran from inside the build directory.
+
+#### Building Black Magic Debug Firmware
+
+To build the firmware you need to configure the probe hardware the firmware will run on, as well as a cross-compilation toolchain used.
+
+For convenience, a cross-file for each supported hardware probe is available which provides a sane default configuration for it.
+
+The build configuration command for the native probe may look like:
+``` sh
+meson setup build --cross-file cross-file/native.ini
+```
+
+Alternatively (for advanced users), if you wish to configure manually, for instance while writing support for a new probe, or a different toolchain, you can run:
+``` sh
+meson setup build --cross-file cross-file/arm-none-eabi.ini -Dprobe=native
+```
+
+You now should have a `build` directory from where you can build the firmware, this is also where your binary files will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
+
+Ypu should now see the resulting binaries in `build`, in this case:
+```
+blackmagic_native_firmware.bin
+blackmagic_native_firmware.elf
+```
+
+These are the binary files you will use to flash to your probe.
+
+##### region `rom' overflowed
+
+It may happen, while working with non default configurations or the project latest version from git, that the firmware does not fit in the available space for the configured probe, this could look something like:
+```
+arm-none-eabi/bin/ld: region `rom' overflowed by 4088 bytes
+Memory region         Used Size  Region Size  %age Used
+             rom:      135160 B       128 KB    103.12%
+             ram:        5708 B        20 KB     27.87%
+collect2: error: ld returned 1 exit status
+```
+
+This is not unexpected, as some hardware probe have limited space for firmware, you can get around it by disabling some features or target support:
+``` sh
+meson setup --reconfigure -Dtargets=cortexm,stm -Drtt_support=false
+```
+
+#### Building Black Magic Debug App
+
+The Black Magic Debug App (BMDA) is always build by default, even for firmware builds, so long as all it's dependencies are found, you can find the executable under the build directory, named simply `blackmagic`
+
+If you wish to build only the BMDA, you can set the hardware `probe` option to an empty string `-Dprobe=''`, this is the default value:
+``` sh
+meson setup build
+```
+
+You now should have a `build` directory from where you can build the app, this is also where your executable will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
+
+Ypu should now see the resulting executable in `build`:
+```
+blackmagic
+```
+
+Most hosted building is done on and for Linux. BMP-hosted for windows can also be build with Mingw on Linux.
```

Comment:
This line can be dropped as it's wrong and has been for ages. BMDA on Windows (the proper name for it) should be done with MSYS2, and we do the full spread of builds for all 3 big name OSes so they're all first-party supported now.

If MinGW were to be named it's purely as a curious extra method. We'd ideally prefer users to use `xwin` to get the headers and use Clang-cl to do a UCRT64 cross build if going down this route. We can provide a cross-file in a follow-up PR that provides the proper environment for doing this as we wrote a Meson configuration for this for bmpflash cross-builds.

Reply:
Ah, I just copied it over without paying any attention to it, I'm unfamiliar with the windows workflow anyway. removed.